### PR TITLE
ADEN-10604 | Add special features to modal

### PR DESCRIPTION
**Pull request by Loczi94**

### CODE REVIEW

File: src/gdpr/ConsentManagementProvider.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const getDefaultCookieAttributes = () => ({
 const getDefaultOptions = () => ({
     allowedVendors: null,
     allowedVendorPurposes: null,
+    allowedSpecialFeatures: null,
```
Comment (dmnsobczak) [inline src/gdpr/ConsentManagementProvider.js:29]: We can get rid of allowing non-vendor GVL elements

File: src/gdpr/ConsentManagementProvider.js [hunk 2]
```diff
@@ -193,22 +194,23 @@ class ConsentManagementProvider {
...
+        tcModel.specialFeatureOptins.set(Array.isArray(allowedSpecialFeatures) ? allowedSpecialFeatures : []);
         tcModel.vendorConsents.set(Array.isArray(allowedVendors) ? allowedVendors : []);
         // ToDo: proper implementation of Right to Object
         tcModel.purposeLegitimateInterests.set(Array.isArray(allowedVendorPurposes) ? allowedVendorPurposes : []);
         tcModel.vendorLegitimateInterests.set(Array.isArray(allowedVendors) ? allowedVendors : []);
         // ToDo: figure out the proper value
         // tcModel.publisherCountryCode();
 
-        debug('GDPR', 'Consent saved with vendors: ', allowedVendors, ' and purposes', allowedVendorPurposes);
+        debug('GDPR', 'Consent saved with vendors: ', allowedVendors, ' and purposes', allowedVendorPurposes, 'and special feature options', allowedSpecialFeatures);
```
Comment (dmnsobczak) [inline src/gdpr/ConsentManagementProvider.js:213]: `'and special` => `' and special`

File: src/gdpr/TrackingOptIn.js [hunk 3]
```diff
@@ -210,6 +214,7 @@ class TrackingOptIn {
             // ToDo: get rid of hardcoded list of purposes during cleanup
             enabledPurposes: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
             enabledVendors: this.options.enabledVendors,
+            enabledSpecialFeatures: [1, 2],
```
Comment (dmnsobczak) [inline src/gdpr/TrackingOptIn.js:217]: Same ToDo as in L214

File: src/modal/Preferences-test.js [hunk 4]
```diff
@@ -24,6 +26,11 @@ function updatePurposes(consentedVendors, consentedPurposes) {
     mockData.consentedVendors = consentedVendors;
 }
 
+function updateSpecialFeatures(consentedVendors, consentedSpecialFeatures) {
```
Comment (dmnsobczak) [inline src/modal/Preferences-test.js:29]: 💎

File: src/modal/Preferences.js [hunk 5]
```diff
@@ -120,32 +138,58 @@ class Preferences extends Component {
...
-        const { consentedPurposes, consentedVendors, content, tracker } = this.props;
+        const { consentedPurposes, consentedSpecialFeatures, consentedVendors, content, tracker } = this.props;
 
         return purposes.map((purpose) => (
             <PreferencesSection
                 allFeatures={this.state.features}
                 allFeaturesSpecial={this.state.specialFeatures}
                 allPurposes={purposes}
                 allPurposesSpecial={this.state.specialPurposes}
+                allItems={purposes}
```
Comment (dmnsobczak) [inline src/modal/Preferences.js:153]: I don't understand why we're passing purposes object two times?

**Review by dmnsobczak — APPROVED** [review]

File: src/modal/Preferences.js (hunk 5 above)
Comment (Loczi94) [reply src/modal/Preferences.js:153]: It's needed to reuse `PreferencesSection` component to render both: Our Partners' Purposes Section and Our Partners' Special Features Section and to be able to pass purposes and specialFeatures to `PreferencesVendorList` component.

File: src/gdpr/TrackingOptIn.js (hunk 3 above)
Comment (dmnsobczak) [reply src/gdpr/TrackingOptIn.js:165]: Will do another time

File: src/gdpr/ConsentManagementProvider.js (hunk 1 above)
Comment (dmnsobczak) [reply src/gdpr/ConsentManagementProvider.js:29]: Next sprint

**Review by Loczi94 — COMMENTED** [review]
Clean up changes by @dmnsobczak approved. ✅

**Pull request merged**